From 97127c3bec87b41585c7c6a99e09a14f182ae1f2 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Thu, 23 May 2024 09:41:31 +0200
Subject: [PATCH 0973/1695] clk: stm32mp25: determine rate return error on
 unsupported frequency

When the functions clk_cpu1_determine_rate() and clk_stm32_pll1_set_rate()
search in the list of supported frequency for the associated OPP defined
in device tree, the function should return an error to indicate that the
requested OPP is not defined.

This patch is a preliminary step to skip OPP when the associated
PLL settings is not provided to the RCC.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 12e3630b1fb5183b4174b176213c234ba5089119 clk: stm32mp25: determine rate return error on unsupported frequency]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I71426006764ac73b49a6c0686f0f3f0bb1a36079
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/382032
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/clk/clk-stm32mp25.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/drivers/clk/clk-stm32mp25.c b/core/drivers/clk/clk-stm32mp25.c
index 8898e4416..0eb41ec46 100644
--- a/core/drivers/clk/clk-stm32mp25.c
+++ b/core/drivers/clk/clk-stm32mp25.c
@@ -2570,7 +2570,7 @@ static TEE_Result clk_stm32_pll1_set_rate(struct clk *clk __unused,
 
 	opp = clk_stm32_get_opp_config(pdata->opp->cpu1_opp, rate);
 	if (!opp)
-		return TEE_ERROR_GENERIC;
+		return TEE_ERROR_ITEM_NOT_FOUND;
 
 	pll_conf = &opp->pll_cfg;
 
@@ -2999,7 +2999,7 @@ static TEE_Result clk_cpu1_determine_rate(struct clk *clk,
 
 	opp = clk_stm32_get_opp_config(pdata->opp->cpu1_opp, rate);
 	if (!opp)
-		return TEE_SUCCESS;
+		return TEE_ERROR_ITEM_NOT_FOUND;
 
 	index = (opp->src & MUX_SEL_MASK) >> MUX_SEL_SHIFT;
 
-- 
2.47.2

